Tests/test_imp.py

```diff
@@ -13,1344 +13,1318 @@
+            def __init__(self, name): self.__name__ = name
+            def __repr__(self): return 'MockModule("' + self.__name__ + '")'
+
+        TopModule = MockModule("TopModule")
+        sys.modules["TopModule"] = TopModule
+        
+        SubModule = MockModule("SubModule")
+        theObj = object()
+        SubModule.Object = theObj
+        TopModule.SubModule = SubModule
+        sys.modules["TopModule.SubModule"] = SubModule
+        
+        # clear the existing names from our namespace...
+        x, y = TopModule, SubModule
+        del TopModule, SubModule
+        
+        # verify we can import TopModule w/ TopModule.SubModule name
+        import TopModule.SubModule
+        self.assertEqual(TopModule, x)
+        self.assertTrue('SubModule' not in dir())
             
-# test init_frozen
-def test_init_frozen():
-    for name in temp_name:
-        f = imp.init_frozen(name)
-        if f != None :
-            Fail("return object should be None!")
-    
-# is_builtin
-def test_is_builtin():
-   
-    AreEqual(imp.is_builtin("xxx"),0)
-    AreEqual(imp.is_builtin("12324"),0)
-    AreEqual(imp.is_builtin("&*^^"),0)
-    
-    AreEqual(imp.is_builtin("dir"),0)
-    AreEqual(imp.is_builtin("__doc__"),0)
-    AreEqual(imp.is_builtin("__name__"),0)
-    
-    AreEqual(imp.is_builtin("_locle"),0)
-    
-    AreEqual(imp.is_builtin("cPickle"),1)
-    AreEqual(imp.is_builtin("_random"),1)
+        # verify we can import Object from TopModule.SubModule
+        from TopModule.SubModule import Object
+        self.assertEqual(Object, theObj)
         
-    # nt module disabled in Silverlight
-    if is_posix:
-        AreEqual(imp.is_builtin("posix"),1)
-    else:
-        AreEqual(imp.is_builtin("nt"),1)
+        # verify we short-circuit the lookup in TopModule if
+        # we have a sys.modules entry...
+        SubModule2 = MockModule("SubModule2")
+        SubModule2.Object2 = theObj
+        sys.modules["TopModule.SubModule"] = SubModule2
+        from TopModule.SubModule import Object2
+        self.assertEqual(Object2, theObj)
         
-    AreEqual(imp.is_builtin("thread"),1)
+        del sys.modules['TopModule']
+        del sys.modules['TopModule.SubModule']
     
-    
-    # there are a several differences between ironpython and cpython
-    if is_cli:
-        AreEqual(imp.is_builtin("copy_reg"),1)
-    else:
-        AreEqual(imp.is_builtin("copy_reg"),0)
-    
-    # supposedly you can't re-init these
-    AreEqual(imp.is_builtin("sys"), -1)
-    AreEqual(imp.is_builtin("__builtin__"), -1)
-    AreEqual(imp.is_builtin("exceptions"), -1)
-    
-    imp.init_builtin("sys")
-    imp.init_builtin("__builtin__")
-    imp.init_builtin("exceptions")
-
-@skip("win32", "multiple_execute", "stdlib")
-def test_sys_path_none_builtins():
-    prevPath = sys.path
-
-    #import some builtin modules not previously imported
-    try:
-        sys.path = [None] + prevPath
-        Assert('datetime' not in sys.modules.keys())
```

Comment:
Yes, unittest will import datetime (at some level), so datetime is in sys.modules since we are now using unittest.